### PR TITLE
Extend `Dropdown::ToggleButton` and update `Dropdown::ToggleIcon` styles

### PR DESCRIPTION
**Pull request by alex-ju**
### :pushpin: Summary

Extend `Hds::Dropdown::ToggleButton` with support for leading `Icon`, `Badge`, and `BadgeCount` and update  `Dropdown::ToggleIcon` styles to support the upcoming 'Filter' pattern.

### :hammer_and_wrench: Detailed description

 - Update `ToggleIcon`'s default state style to align with the style of the secondary button. This improves the affordance of the button and allows us to future-proof for a dark mode version. The other states remain the same.
 - Allow `Badge` (via `@badge` and `@badgeIcon`) and `BadgeCount` (via `@count`) as trailing content in `Button` and `ToggleButton`
 - Allow Icon (via `@icon`) alongside chevron in `ToggleButton` (we're introducing a `@hasChevron` argument to the Button component, for internal usage, to allow the `ToggleButton` to also have a leading icon when needed)

[👉 Preview `Button` showcase](https://hds-showcase-93qco6btu-hashicorp.vercel.app/components/button)
[👉 Preview `Dropdown::ToggleButton` showcase](https://hds-showcase-93qco6btu-hashicorp.vercel.app/components/dropdown)

#### Notes on tradeoffs

To avoid breaking changes on the Button component:
- we're taking the approach of generating the Badge within the Button component and opening the API only for the text and icon of the Badge (e.g. `<Hds::Button @text="button" @badge="badge"/>`), as opposed to allowing users to invoke a Badge instance within the Button component (e.g. `<Hds::Button>button <Hds::Badge @text="badge"></Hds::Button>`) – this also ensures the required adjacent contrast required for accessibility compliance
- we're adding the `@hasChevron` as a workaround to allow a leading icon and the chevron as trailing icon on a Button, as opposed to changing the Button API to allow both a leading and a trailing icon.

### :link: External links

Jira ticket: [HDS-1109](https://hashicorp.atlassian.net/browse/HDS-1109)
Figma file: https://www.figma.com/file/Kgc8BgB8jGuJXISndgTrD1/Dropdown?node-id=29345%3A69392&t=h2KjNpimGUUm0DCW-4 and https://www.figma.com/file/SBw7wkwlPBnkNERKRAMxoR/Badge-in-Button?node-id=30893%3A117893&t=t6G71WRPZBKwfjhr-4

***

### 👀 Reviewer's checklist:

- [ ] +1 Percy if applicable
- [ ] Confirm that PR has a changelog update via [Changesets](https://github.com/changesets/changesets) if needed

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

[HDS-1109]: https://hashicorp.atlassian.net/browse/HDS-1109?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by didoo — COMMENTED** [review]
Added a few comments for now. 
I want to discuss with you the general approach, though. I'll set up a meeting for later today.

File: packages/components/addon/components/hds/button/index.hbs [hunk 1]
```diff
@@ -20,25 +20,39 @@
...
+        @size="small"
+        @type={{if (not-eq this.color "primary") "inverted"}}
+      />
+    {{/if}}
+    {{#if (and this.icon (eq this.iconPosition "trailing"))}}
+      <div class="hds-button__icon">
+        <FlightIcon @name={{this.icon}} @size={{this.iconSize}} @stretched={{true}} />
+      </div>
+    {{/if}}
+    {{#if @hasChevron}}
```
[the remark this answers is not in the document]
Comment (alex-ju) [reply packages/components/addon/components/hds/button/index.hbs:53]: Having a separate component for the `Dropdown::Toggle` becomes the preferred option as, after syncing with design on Friday, we seem to aim for a different badge design for the Toggle (compared to button) as well.

**Comment by alex-ju** [comment]
For visibility – we're planning to split this work into `ToggleButton` and `Button` updates. The `ToggleButton` will require a bit of refactoring, so I'll pivot the work in this PR

**Comment by alex-ju** [comment]
On second thought, starting from scratch on this makes more sense, as most of this is not relevant anymore.

**Pull request closed without merging**